sdk/search/azure-search-documents/azure/search/documents/_headers_mixin.py

```diff
@@ -4,20 +4,15 @@
 # license information.
 # --------------------------------------------------------------------------
 
-from typing import TYPE_CHECKING
+from typing import Optional, Dict
 
-if TYPE_CHECKING:
-    from typing import Optional
 
-
-class HeadersMixin(object):
+class HeadersMixin:
     @property
-    def _headers(self):
-        # type() -> dict
+    def _headers(self) -> Dict:
         return {"api-key": self._credential.key, "Accept": self._ODATA_ACCEPT}
 
-    def _merge_client_headers(self, headers):
-        # type(Optional[dict]) -> dict
+    def _merge_client_headers(self, headers: Optional[Dict]) -> Dict:
```

Comment:
This one is interesting, since headers are defined as `Dict[str, str]` in azure.core the method can be `def _merge_client_headers(self, headers: Optional[Dict[str, str]) -> Dict:`.

Does the return also change to Dict[str, str] given that an empty dict can also return ?